frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/SlideInUnmanagedBranch.java

```diff
@@ -0,0 +1,88 @@
+package com.virtuslab.gitmachete.frontend.ui.impl.table;
+
+import static com.intellij.openapi.application.ModalityState.NON_MODAL;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import java.nio.file.Path;
+import java.util.function.Consumer;
+
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsNotifier;
+import com.intellij.util.ModalityUiUtil;
+import io.vavr.control.Try;
+import lombok.AllArgsConstructor;
+import lombok.CustomLog;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+
+import com.virtuslab.binding.RuntimeBinding;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRepositoryCache;
+import com.virtuslab.gitmachete.backend.api.IManagedBranchSnapshot;
+import com.virtuslab.gitmachete.frontend.ui.api.gitrepositoryselection.IGitRepositorySelectionProvider;
+import com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod(GitVfsUtils.class)
+@AllArgsConstructor
+@CustomLog
+public class SlideInUnmanagedBranch {
+
+  private final Project project;
+  private final String branchName;
+  private final IGitRepositorySelectionProvider gitRepositorySelectionProvider;
+  private final Consumer<Path> onFailurePathConsumer;
+  private final Consumer<String> onSuccessInferredParentBranchNameConsumer;
+
+  public void enqueue(Path macheteFilePath) {
+    LOG.info("Enqueuing unmanaged branch notification");
+    val selectedRepository = gitRepositorySelectionProvider.getSelectedGitRepository();
+    if (selectedRepository == null) {
+      LOG.error("Can't notify about unmanaged branch because of undefined selected repository");
+      return;
+    }
+    Path rootDirPath = selectedRepository.getRootDirectoryPath().toAbsolutePath();
+    Path mainGitDirPath = selectedRepository.getMainGitDirectoryPath().toAbsolutePath();
+    Path worktreeGitDirPath = selectedRepository.getWorktreeGitDirectoryPath().toAbsolutePath();
+
+    new Task.Backgroundable(project,
+        getString("string.GitMachete.EnhancedGraphTable.unmanaged-branch-notification.task-title")) {
+      @Override
+      @UIThreadUnsafe
+      public void run(ProgressIndicator indicator) {
+        LOG.debug("Running infer parent for unmanaged branch slide in task");
+        val inferParentRunResult = Try.of(() -> {
+          val gitMacheteRepository = RuntimeBinding.instantiateSoleImplementingClass(IGitMacheteRepositoryCache.class)
+              .getInstance(rootDirPath, mainGitDirPath, worktreeGitDirPath);
+          val managedBranches = gitMacheteRepository.discoverLayoutAndCreateSnapshot().getManagedBranches();
```

Comment:
Hmmm don't use the **discovered** layout  (`discoverLayoutAndCreateSnapshot`) here, use the **current** layout as seen in .git/machete. See `com.virtuslab.gitmachete.frontend.ui.impl.table.GitMacheteRepositoryUpdateBackgroundable#updateRepositorySnapshot` -> `com.virtuslab.gitmachete.backend.api.IGitMacheteRepository#createSnapshotForLayout`.

The problem with using discovered layout is that even if the inferred parent might be present among discovered layout's branches... it doesn't matter that it'll be present among current layout branches (in fact, that's how I noticed that _sum ting wong_ when testing a snapshot version from this PR).

Reply:
oh, yeah. good point. that was my first a bit hackish halfway approach. and I see why I did it. the problem is that in order to `inferParentForLocalBranch` I need `IGitMacheteRepository` but until now we have no access to it once the snapshot is created. 

IIRC we ve introduced repository snapshot to keep things more immutable.

anyway, ill try to rework that

Reply:
> IIRC we ve introduced repository snapshot to keep things more immutable.

Or even actually, to keep the separation between access to **mutable state** of the repo (via `GitMacheteRepository`) and an **immutable snapshot** of that state (`GitMacheteRepositorySnapshot`)

Reply:
> anyway, ill try to rework that

Maybe passing a `GitMacheteRepository` to this action won't be the worst idea 🤔